### PR TITLE
grab detailed champ object, random QWE, Ability Image

### PR DESCRIPTION
**Pull request by brycerobinsn**
- grab SelectedRandomChampion detailed object
- add Ability image
- randomly select QWE

### CODE REVIEW

**Comment by brycerobinsn** [comment]
https://github.com/cannedyeti/ultimate-bravery-v3/assets/80009651/942c636e-8860-41df-890e-2500cf51567b

File: src/api/fetchRiotData.js [hunk 1]
```diff
@@ -4,9 +4,19 @@ export const CHAMPION_IMAGE_BASE_URL =
...
   ).then((x) => x.json());
   return data;
 };
+
+export const getChampionData = async (championName) => {
+  const data = await fetch(
+    `https://ddragon.leagueoflegends.com/cdn/14.10.1/data/en_US/champion/${championName}.json`
+  ).then((x) => x.json());
+  return data;
+}
```
Comment (cannedyeti) [inline src/api/fetchRiotData.js]: Good habit to make sure you have an empty line at the end of your files.

File: src/common/helpers.js [hunk 2]
```diff
@@ -7,3 +7,7 @@ export const randomUniqueItemsFromArray = (arr, numItems) => {
   const shuffled = arr.sort(() => 0.5 - Math.random());
   return shuffled.slice(0, numItems);
 };
+
+export function getRandomInt(max) {
+  return Math.floor(Math.random() * max);
+}
```
Comment (cannedyeti) [inline src/common/helpers.js]: Missing empty line at end

File: index.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-<!DOCTYPE html>
+<!doctype html>
```
Comment (cannedyeti) [inline index.html:1]: What's happening here 😆

File: src/features/bravery/components/RandomItems.jsx [hunk 4]
```diff
@@ -1,18 +1,27 @@
...
+import { ITEM_IMAGE_BASE_URL, SPELL_IMAGE_BASE_URL } from "src/api/fetchRiotData";
+import { getRandomInt } from "src/common/helpers";
 import { BraveryContext } from "src/features/bravery/components/BraveryReducer";
 
 export function RandomItems() {
   const { state } = useContext(BraveryContext);
-
-  console.log(state.selectedRandomChampion);
+  const randy = getRandomInt(3);
+  const qwe = ["Q", "W", "E"]
```
Comment (cannedyeti) [inline src/features/bravery/components/RandomItems.jsx]: probably worth putting this in the `./features/bravery/components/helpers.js` file. 
Also as much as I love `randy` it's better to get into good naming habits (hardest part of programming) lmao. 
Maybe something like `randomAbilityIndex` and `abilityArray`

File: src/features/bravery/components/RandomItems.jsx [hunk 5]
```diff
@@ -1,18 +1,27 @@
...
-      <Box>Max Q</Box>
+      <Box>{qwe[randy]}
+      <Tooltip label={state.selectedRandomChampion?.spells[randy].name}>
+      <Avatar
+        src={`${SPELL_IMAGE_BASE_URL}${ability}`}
+        borderRadius={2}
+      />
+      </Tooltip>
+      </Box>
+
```
Comment (cannedyeti) [inline src/features/bravery/components/RandomItems.jsx]: Fix your linter so it doesnt mess up whitespace! 😠

**Review by cannedyeti — APPROVED** [review]
Couple nits, but looks good to me :)
May be worth trying to do something so we don't have to call the api every time, but that's likely on the fake DB end.

**Comment by cannedyeti** [comment]
Actually, would you mind throwing a [Chakra spinner](https://v2.chakra-ui.com/docs/components/spinner) that renders while the api is getting fetched. Having a user avatar as the loading buffer makes it a bit weird. 

Could look something like this 

``` 
{ ability?.name ? 
  <Tooltip >
      <Avatar /> 
  </Tooltip>
   : 
  <Spinner />
} 
```

File: src/features/bravery/components/RandomItems.jsx [hunk 6]
```diff
@@ -1,18 +1,27 @@
...
+import { getRandomInt } from "src/common/helpers";
 import { BraveryContext } from "src/features/bravery/components/BraveryReducer";
 
 export function RandomItems() {
   const { state } = useContext(BraveryContext);
-
-  console.log(state.selectedRandomChampion);
+  const randy = getRandomInt(3);
+  const qwe = ["Q", "W", "E"]
+  const ability = state.selectedRandomChampion?.spells[randy].image.full;
```
Comment (cannedyeti) [inline src/features/bravery/components/RandomItems.jsx]: Could change this to just ` state.selectedRandomChampion?.spells[randy]` that way yo ucan use `ability` for the image and the name with less bloat.

**Comment by cannedyeti** [comment]
Also just noticed that deselecting/selecting champions when an ability is selected will rerun random ability. May need to add the ability to the context reducer.

File: index.html (hunk 3 above)
Comment (brycerobinsn) [reply index.html:1]: I have no idea how this happened lol

**Comment by cannedyeti** [comment]
I was able to add the full champ data to the original championData object. Shouldn't be a need for a second api call just for the ability anymore.

**Comment by brycerobinsn** [comment]
feels a bit overkill but added ability and index to the reducer for now

**Pull request merged**